context.go

```diff
@@ -24,6 +28,14 @@ type Context struct {
 // NewContext creates a new context. For use in when invoking an App or Command action.
 func NewContext(app *App, set *flag.FlagSet, parentCtx *Context) *Context {
 	c := &Context{App: app, flagSet: set, parentContext: parentCtx}
+	ctx, cancel := context.WithCancel(context.Background())
+	go func() {
+		defer cancel()
+		sigs := make(chan os.Signal, 1)
+		signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
+		<-sigs
+	}()
+	c.Context = ctx
```

Comment:
If parent context is not nil, should we not inherit? Also a bit grim the client can't inject a context from higher up, so he could use it to cancel his own commands.

Reply:
@AudriusButkevicius the client can wrap the context if they wish to. So that should be possible. Note that net/http does the same thing with the `*http.Request.Context()` in that it automatically includes cancellation if the client cancels the request.

Reply:
Then we should leave it empty by default, or context.Background()?

Reply:
My question is whether we should create a new context for every subcommand or just pass the same context down.

Reply:
@AudriusButkevicius I see. Is there ever a situation where the CLI runs multiple subcommands at the same time? Or does urfave/cli only run one sub command per binary invocation? If the latter, then we're never gonna create multiple contexts. 

That said, if you can point to any changes, I'm happy to make those :+1:

Reply:
We run NewContext for every subcommand, as we traverse down the command tree:
https://github.com/urfave/cli/blob/7675649a174ac724b1d7fa9d5932eff3dd1582a2/app.go#L323

So for `command subcomand-a subcommand-a-b subcommand-a-b-c` we'd init 4 contexts, all of which would set up their own signal handlers and have nothing in common with the parent context, hence cancellation at the root would do nothing I think.

Reply:
@AudriusButkevicius sounds good. Thanks for the clarification. I will verify and let you know :+1:

Reply:
@AudriusButkevicius ok i just updated the PR to only run one signal handler, and all sub-commands will inherit that context. 

Interestingly, `command sub-a sub-a-b sub-a-b-c` will call NewContext on each one as you pointed out, but the `Action` function that will run is only `sub-a-b-c` which makes me think that maybe urfave/cli should not create multiple contexts? It should probably just create one context for the Action it will run. 

If I'm not missing something, that's a topic for another PR. For now let me know if this is good to merge.

Thanks!